daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -695,9 +696,13 @@ object Main {
       case InfosetType.W3CDOM => Left(new W3CDOMInfosetOutputter())
       case InfosetType.SAX => Right(new DaffodilParseOutputStreamContentHandler(os, pretty=true))
       case InfosetType.EXI => {
-        val exiFactory = DefaultEXIFactory.newInstance()
-        exiFactory.getFidelityOptions.setFidelity(FidelityOptions.FEATURE_PREFIX, true)
-        val exiResult = new EXIResult()
+        val exiFactory = DefaultEXIFactory.newInstance
+        if (schemaOpt.isDefined) {
+          val grammarFactory = GrammarFactory.newInstance
+          val grammar = grammarFactory.createGrammars(schemaOpt.get.toURL.openStream(), DFDLCatalogResolver.get)
```

Comment:
The `getInfosetOutputter` function is called for every iteration of a performance run. I imagine building this grammar is pretty expensive, so should probably be down outside the main loop and passed into this function when EXI is needed. In fact, maybe we should create the entire EXIFactory outside of the main loop and pass that into this function, so that this function only creates a new EXIResult/handler?

Reply:
I didn't realize that getInfosetOutputter gets called on every run  Having the factory set up once will also help with clarity for unparsing as well as there will only be one EXIFactory created that can be used for both parsing and unparsing.